File: sample/src/main/java/com/surf/myapplication/interactor/location/LocationData.java
package com.surf.myapplication.interactor.location;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class LocationData {
    private String provider;
    private double lon;
    private double lat;
    private Long time;

    public static LocationData empty() {
        return new LocationData("Unknown", 0, 0, 0L);
    }
}

File: location/src/main/java/ru/surfstudio/android/location/LocationData.kt
package ru.surfstudio.android.location

/**
 * Пакет данных геолокации
 */
const val UNKNOWN_PROVIDER = "Unknown"
val UNKNOWN_LOCATION = LocationData(UNKNOWN_PROVIDER, 0.0, 0.0, 0)

data class LocationData(val provider: String,
                        val lon: Double,
                        val lat: Double,
                        val time: Long) {
    fun isProviderUnknown() = provider == UNKNOWN_PROVIDER
}
